Bottom: 9c56acd0a8c26c2206bec1abe44f11ddb00b0ab7
Top:    e809f70c265b9c91183bdb6f160d416d800fbb13
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-10-26 13:47:51 +0000

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index 919a317..cc3f718 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -227,6 +227,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
